lambdas/call-dcmaw-async-cri/src/test/java/uk/gov/di/ipv/core/calldcmawasynccri/service/DcmawAsyncCriServiceTest.java

```diff
@@ -111,15 +119,21 @@ void startDcmawAsyncSession_WhenCalled_ReturnsAVc() throws Exception {
                                                 && JOURNEY_ID.equals(crbd.getJourneyId())
                                                 && CRI_CLIENT_ID.equals(crbd.getClientId())
                                                 && CRI_OAUTH_STATE.equals(crbd.getState())
-                                                && REDIRECT_URL.equals(crbd.getRedirectUri()))))
+                                                && Objects.equals(
+                                                        expectedRedirectUrl,
+                                                        crbd.getRedirectUri()))))
                 .thenReturn(vcResponse);
 
         // Act
         var response =
                 dcmawAsyncCriService.startDcmawAsyncSession(
-                        CRI_OAUTH_STATE, clientOAuthSessionItem, ipvSessionItem);
+                        CRI_OAUTH_STATE, clientOAuthSessionItem, ipvSessionItem, context);
 
         // Assert
         assertEquals(vcResponse, response);
     }
+
+    private static Stream<Arguments> contextsAndClientCallbackUrls() {
+        return Stream.of(Arguments.of(MAM_CONTEXT, REDIRECT_URL), Arguments.of(DAD_CONTEXT, null));
```

Comment:
These unit tests should eventually be paired with contract tests with dcmaw async